src/pages/Auction/index.tsx

```
@@ -144,39 +144,32 @@ const AuctionPage = ({ data: { auctionIdentifier, derivedAuctionInfo, graphInfo
   const navigate = useNavigate()
   const invalidAuction = !loading && (!auctionIdentifier || derivedAuctionInfo === undefined)
   const [pageViews, setPageViews] = useState()
-  const [path, setPath] = useState('')
 
   useEffect(() => {
     const domain = window.location.origin
-    const pagePath = window.location.pathname
-    setPath(pagePath)
+    const path = window.location.pathname
     const getPageViews = async () => {
       try {
         const value = await fetch(`${domain}/api/viewInfo/pageViews?path=${path}`)
-        const valueJson = await value.json()
-        const pageView = valueJson[0].rows[0].metricValues[0].value
-        setPageViews(pageView)
+        if (value.ok) {
+          const valueJson = await value.json()
+          const numberValue = Number(valueJson)
+          console.log(typeof numberValue)
+          if (typeof numberValue === 'number') {
+            setPageViews(valueJson)
+          } else {
+            return
+          }
```

I don't think any of this is necessary. Additionally, if you cast something like `Number('fox')` it returns `NaN`. However, the check that `NaN` is a number passes, because it's considered a number. There shouldn't be a need to have this page views as a number as a string or a number will be fine to display.

---

And from your earlier comment about the route sometimes not returning, if there isn't a value, the endpoint should return an error which would prevent the thing from displaying as well.

That is, if the response is `ok` it should always return what you expect. Although it's good to check what the API returns is good to display, e.g. not undefined, any of that logic ensuring the value is either correct or returns nothing should be in the serverless function, not here.

---

Fixed